stdlib/public/Concurrency/Task.cpp

```diff
@@ -1568,6 +1601,7 @@ static void swift_task_asyncMainDrainQueueImpl() {
   swift_task_donateThreadToGlobalExecutorUntil([](void *context) {
     return *reinterpret_cast<bool*>(context);
   }, &Finished);
+  exit(0);
```

Comment:
You're right, not needed. this is just a leftover from when I didn't yet understand how async main works. I can just switch to async main and this can go away.

(But technically, swift_task_asyncMainDrainQueueImpl has a NORETURN attribute. So it's illegal to return from it.)